### PR TITLE
Improve fake ad for demo purpose

### LINKED ISSUE #13281: Fake Ad enhancement
ref: #9311
We create the `amp-ad-network-fake-impl` extension to test fake Amp Ad locally. 
Right now this extension doesn't work properly and its documentation is outdated.
And the extension only works in localDev mode, and thus makes it impossible to demo an AMPHTML ad creative using the extension.

- [x] Fix the extension, right now when serving creative from `amp-ad-network-fake-impl/data/` folder, the creative always render in 3p case.
- [x] make the extension only send ad request with invalid id and remove check to localDev mode

cc @lannka @taymonbeal @glevitzky @keithwrightbos @jonkeller

For #13283

### PR DESCRIPTION
**Pull request by zhouyx**
closes #13281

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -1290,7 +1282,8 @@ export class AmpA4A extends AMP.BaseElement {
     let renderPromise = Promise.resolve(false);
     if ((method == XORIGIN_MODE.SAFEFRAME ||
          method == XORIGIN_MODE.NAMEFRAME) &&
-        this.creativeBody_) {
+        this.creativeBody_ &&
+        this.element.getAttribute('type') != 'fake') {
```
Comment (lannka) [inline extensions/amp-a4a/0.1/amp-a4a.js]: as discussed offline, pls check if we can get rid of this line

File: extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js [hunk 2]
```diff
@@ -41,17 +38,20 @@ export class AmpAdNetworkFakeImpl extends AmpA4A {
...
   /** @override */
   isValidElement() {
-    // Note: true is the default, so this method is not strictly needed here.
-    // But a network implementation might choose to implement a real check
-    // in this method.
+    // To send out ad request, ad type='fake' requires the id set to an invalid
+    // value start with `i-amphtml-demo`. So that fake ad can only be used in
+    // invalid AMP pages.
+    const id = this.element.getAttribute('id');
+    if (!id || !startsWith(id, 'i-amphtml-demo')) {
```
Comment (lannka) [inline extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js]: `i-amphtml-demo-`

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 1 above)
Comment (zhouyx) [reply extensions/amp-a4a/0.1/amp-a4a.js]: removed. Filed bug #13326

File: extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js (hunk 2 above)
Comment (zhouyx) [reply extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js]: done

File: examples/ampcontext-creative-json.html [hunk 3]
```diff
@@ -1,3 +1 @@
-{"creative":
-"<!doctype html><html>  <head>    <style>body { background: lightblue; }</style>    <script src='https://localhost:8000/examples/ac-creative.js'></script>  </head>  <body>    Test Ad using ampcontext.js to create window.context    <button onclick='toggleObserveIntersection()'>Toggle Observe Intersections</button>    <button onclick='toggleObserveVisibility()'>Toggle Observe visibility</button>    <button onclick='window.context.requestResize(500,600)'>resize ad</button>    <button onclick='console.log(window.context)'>console.log(window.context)</button>  </body></html>",
-"signature":"123"}
+<!doctype html><html>  <head>    <style>body { background: lightblue; }</style>    <script src='https://localhost:8000/examples/ac-creative.js'></script>  </head>  <body>    Test Ad using ampcontext.js to create window.context    <button onclick='toggleObserveIntersection()'>Toggle Observe Intersections</button>    <button onclick='toggleObserveVisibility()'>Toggle Observe visibility</button>    <button onclick='window.context.requestResize(500,600)'>resize ad</button>    <button onclick='console.log(window.context)'>console.log(window.context)</button>  </body></html>
```
Comment (lannka) [inline examples/ampcontext-creative-json.html]: can you format the file?

File: extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js [hunk 4]
```diff
@@ -62,35 +62,28 @@ export class AmpAdNetworkFakeImpl extends AmpA4A {
...
-          return response.text().then(
-              responseText => new Response(
-                  this.transformCreativeLocalDev_(responseText),
-                  {status, headers}));
-        }
+
+      // In the convert creative mode the content is the plain AMP HTML.
+      // This mode is primarily used for A4A Envelop for testing.
+      // See DEVELOPING.md for more info.
+      if (this.element.getAttribute('convert') == 'true') {
```
Comment (lannka) [inline extensions/amp-ad-network-fake-impl/0.1/amp-ad-network-fake-impl.js]: `convert` => `a4a-conversion`

**Comment by zhouyx** [comment]
@keithwrightbos @taymonbeal PTAL and let me know WDYT? Thanks!

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 5]
```diff
@@ -820,9 +811,10 @@ export class AmpA4A extends AMP.BaseElement {
             }))
         .then(status => {
           checkStillCurrent();
-          if (getMode().localDev &&
-              this.element.getAttribute('type') == 'fake') {
-            // do not verify signature for fake type ad
+          if (this.element.getAttribute('type') == 'fake' &&
```
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: We might as well move this before the signatureVerifierFor promise given we don't actually care about the result.

Comment (zhouyx) [reply extensions/amp-a4a/0.1/amp-a4a.js]: Moved.

**Pull request merged**